### ISSUE TITLE
version 0.3.7 seems broken

### ISSUE THREAD

**Issue by skaji** [Raku/tap-harness6#57]
Steps to reproduce:

```console
❯ raku -v
Welcome to Rakudo™ v2022.07.
Implementing the Raku® Programming Language v6.d.
Built on MoarVM version 2022.07.

❯ git clone https://github.com/Raku/tap-harness6

❯ cd tap-harness6

❯ git describe --tags --dirty
0.3.7

❯ echo 'use Test; plan 1; pass "ok";' > test.t

❯ raku -I. -MTAP -e 'TAP::Harness.new.run("test.t").result'
test.t ..1/1Tried to get the result of a broken Promise
  in block <unit> at -e line 1

Original exception:
    No such method 'exit' for invocant of type 'Proc'.  Did you mean
    'emit'?
      in method exit-status at /Users/skaji/tap-harness6/lib/TAP.pm (TAP) line 800
      in method result at /Users/skaji/tap-harness6/lib/TAP.pm (TAP) line 892
      in sub reap-finished at /Users/skaji/tap-harness6/lib/TAP.pm (TAP) line 1060
      in block  at /Users/skaji/tap-harness6/lib/TAP.pm (TAP) line 1039
      in block  at /Users/skaji/tap-harness6/lib/TAP.pm (TAP) line 1028
```

Note that the previous version 0.3.6 works fine:

```console
❯ git checkout 0.3.6

❯ raku -I. -MTAP -e 'TAP::Harness.new.run("test.t").result'
test.t .. ok
All tests successful.
Files=1, Tests=1,  0 wallclock secs
Result: PASS
```

**Comment by Leont** [comment]
Fixed in 0.3.8, sorry about this hickup. I really should add a test for that codepath.

**Comment by skaji** [comment]
@Leont Thanks!
